docs/contributor-guide/how-to/common-rule-scenarios.md

```diff
@@ -0,0 +1,205 @@
+that the thing the rule checks for should exist and be valid/used
+correctly. Examples here are the rules that check for different fields
+of the manifest file. They should not pass if, for example, the web
+manifest file doesn't exist (even if there is a rule that checks
+exactly that).
+
+The recommended way to implement a rule like this is to subscribe
+to the event `scan::end`. If your rule receives that event and has
+not run any validation you should report it.
+
+## Evaluate JavaScript in the page context
+
+Sometimes a rule needs to evaluate some JavaScript in the context of
+the page. To do that you need to use `context.evaluate`. This method
+will always return a `Promise` even if your code does not return one.
+
+One important thing is that your code needs to be wrapped in an
+immediate invoked function expression (IIFE).
+
+The following scripts will work:
+
+```js
+const script =
+`(function() {
+    return true;
+}())`;
+
+context.evaluate(script);
+```
+
+```js
+const script =
+`(function() {
+    return Promise.resolve(true);
+}())`;
+
+context.evaluate(script);
+```
+
+The following does not:
+
+```js
+const script = `return true;`;
+
+context.evaluate(script);
+```
+
+```js
+const script = `return Promise.resolve(true);`;
+
+context.evaluate(script);
+```
+
+## Ignore connectors
+
+If your rule does not work properly with certain connectors you can
+use the property `ignoreConnectors` so it is not run when they are used.
+
+<!-- eslint-disable no-unused-vars, object-curly-newline -->
+
+```js
+const rule = {
+    create(context) {
+        // Your code here
+    },
+
+    meta: {
+        ignoredConnectors: ['jsdom']
+    }
+};
+```
+
+## Interact with other services
+
+You can develop a rule that integrates with other services. `sonarwhal`
+integrates with a few such as `ssllabs`.
+
+Because these online tools usually take a few seconds to return the
+results the guidance is to start the analysis as soon as possible
+and then collect the results as late as possible. This means you
+will have to listen to `scan::start` and `scan::end` events respectively.
+The `create` method of your rule should be similar to the following:
+
+```ts
+create(context: RuleContext): IRule {
+    /** The promise that represents the connection to the online service. */
+    let promise: Promise<any>;
+
+    const start = (data: IScanStartEvent) => {
+        // Initialize promise to service here but do not return it.
+    };
+
+    const end = (data: IScanEndEvent): Promise<any> => {
+        return promise
+            .then((results) => {
+                // Report any results via `context.report` here.
+            })
+            .catch((e) => {
+                // Always good to handle errors.
+            });
+    };
+
+    return {
+        'scan::start': start,
+        'scan::end': end
+    };
+}
+```
+
+In case you need a more complete example, please look at the
+`ssllabs.ts` source code.
```

Comment:
Can you add a link here?

I know the link will be changed soon, but this way we won''t forget to add it, and when it will be broken it will be cached by out script anyway.